units/host_status.go

```diff
@@ -0,0 +1,123 @@
+package units
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/cloud"
+	"github.com/evergreen-ci/evergreen/model/host"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/dependency"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+const cloudStatusJobName = "set-cloud-statuses"
+
+func init() {
+	registry.AddJobType(cloudStatusJobName,
+		func() amboy.Job { return makeManagerJob() })
+}
+
+type cloudStatusJob struct {
+	job.Base `bson:"job_base" json:"job_base" yaml:"job_base"`
+	env      evergreen.Environment
+}
+
+// NewCloudJob gets statuses for all jobs created by Cloud providers which the Cloud providers'
+// APIs have not yet returned all running. It marks the hosts running in the database.
+func NewCloudJob(env evergreen.Environment, id string) amboy.Job {
+	j := makeManagerJob()
+	j.SetID(fmt.Sprintf("%s.%s", cloudStatusJobName, id))
+	j.env = env
+	j.SetPriority(1)
+	return j
+}
+
+func makeManagerJob() *cloudStatusJob {
+	j := &cloudStatusJob{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    cloudStatusJobName,
+				Version: 1,
+			},
+		},
+	}
+
+	j.SetDependency(dependency.NewAlways())
+	return j
+}
+
+func (j *cloudStatusJob) Run(ctx context.Context) {
+	var cancel context.CancelFunc
+	ctx, cancel = context.WithCancel(ctx)
+	defer cancel()
+	defer j.MarkComplete()
+	if j.env == nil {
+		j.env = evergreen.GetEnvironment()
+	}
+
+	hostsToCheck, err := host.Find(host.NotRunning())
+	if err != nil {
+		j.AddError(errors.Wrap(err, "problem finding hosts that are not running"))
+		return
+	}
+	providers := map[string][]host.Host{}
+	for _, h := range hostsToCheck {
+		providers[h.Provider] = append(providers[h.Provider], h)
+	}
```

Comment:
does it make sense to do this in hosts package? potentially using an aggregation to create the map?

Reply:
This is pretty cheap, and an aggregation to create the map would be harder to read, I think.